prow/spyglass/spyglass.go

```diff
@@ -404,7 +404,7 @@ func (sg *Spyglass) RunToPR(src string) (string, string, int, error) {
 func (sg *Spyglass) ExtraLinks(ctx context.Context, src string) ([]ExtraLink, error) {
 	artifacts, err := sg.FetchArtifacts(ctx, src, "", 1000000, []string{prowv1.StartedStatusFile})
 	// Failing to find started.json is okay, just return nothing quietly.
-	if err != nil || len(artifacts) == 0 {
+	if err != nil {
 		logrus.WithError(err).Debugf("Failed to find started.json while looking for extra links.")
```

Comment:
It seems that there are other errors that can be returned from `FetchArtifacts` should it always be saying "Failed to find started.json..."? 
`return arts, fmt.Errorf("error parsing src: %v", err)` < is that also "okay, just return nothing"?
If so we might want to update the comment?

Reply:
For the parsing error I do think that it needs to return error instead of nothing. I wonder if it's intended or not, will check it later.

Imho, the proper fix would be to return error on parsing failure, and return nothing when no artifacts found (with defined error).

Reply:
Done in d56b996, could you review again @MushuEE?